transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/specification/ws/acceptor/LimitsIT.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright 2007-2015, Kaazing Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kaazing.gateway.transport.wsn.specification.ws.acceptor;
+
+import static org.kaazing.test.util.ITUtil.createRuleChain;
+
+import java.net.URI;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+import org.kaazing.gateway.server.test.GatewayRule;
+import org.kaazing.gateway.server.test.config.GatewayConfiguration;
+import org.kaazing.gateway.server.test.config.builder.GatewayConfigurationBuilder;
+import org.kaazing.gateway.transport.IoHandlerAdapter;
+import org.kaazing.gateway.transport.nio.internal.NioSocketAcceptor;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+
+public class LimitsIT {
+    private static String WS_ECHO_SERVICE_ACCEPT_LIMIT = "ws://localhost:8080/echo";
+    private static String WS_PROXY_SERVICE_ACCEPT_LIMIT = "ws://localhost:8010/proxy";
+    private static int PROXY_SERVICE_CONNECT_PORT_LIMIT = 8123;
+    private static String PROXY_SERVICE_CONNECT_LIMIT = "tcp://localhost:" + PROXY_SERVICE_CONNECT_PORT_LIMIT;
+
+    private static String WS_ECHO_SERVICE_ACCEPT_DEFAULT = "ws://localhost:8081/echo";
+    private static String WS_PROXY_SERVICE_ACCEPT_DEFAULT = "ws://localhost:8011/proxy";
+    private static int PROXY_SERVICE_CONNECT_PORT_DEFAULT = 8124;
+    private static String PROXY_SERVICE_CONNECT_DEFAULT = "tcp://localhost:" + PROXY_SERVICE_CONNECT_PORT_DEFAULT;
+    
+    
+    private static int DEFAULT_WEBSOCKET_MAXIMUM_MESSAGE_SIZE = 128*1024;
+    private static int LIMITED_WEBSOCKET_MAXIMUM_MESSAGE_SIZE_125 = 125;
+    public static final long TEST_LIFETIME = TimeUnit.SECONDS.toSeconds(3);
+
+    private final K3poRule k3po = new K3poRule().setScriptRoot("org/kaazing/specification/ws/limits");
+
+    private GatewayRule gateway = new GatewayRule() {
+        {
+            // @formatter:off
+            GatewayConfiguration configuration =
+                    new GatewayConfigurationBuilder()
+                        .service()
+                            .accept(URI.create(WS_PROXY_SERVICE_ACCEPT_LIMIT))
+                            .connect(URI.create(PROXY_SERVICE_CONNECT_LIMIT))
+                            .type("proxy")
```

Comment:
We don't need to test against both proxy and echo services since these tests are purely testing the WebSocket transport. They are not testing services. Just use echo service for simplicity.